app/src/views/tool/ToolLog.vue

```diff
@@ -0,0 +1,136 @@
+        </b-col>
+
+        <b-col>
+          <span v-if="prop.endsWith('_at')">{{ value | readableDate }}</span>
+
+          <div v-else-if="prop === 'suboperations'">
+            <div v-for="operation in value" :key="operation.name">
+              <icon v-if="!operation.success" iname="times" class="text-danger" />
+              <b-link :to="{ name: 'tool-log', params: { name: operation.name } }">
+                {{ operation.description }}
+              </b-link>
+            </div>
+          </div>
+
+          <span v-else>{{ value }}</span>
+        </b-col>
+      </b-row>
+    </card>
+
+    <div v-if="info.error" class="alert alert-danger my-5">
+      <icon iname="exclamation-circle" /> {{ $t('operation_failed_explanation') }}
+    </div>
+
+    <!-- LOGS CARD -->
+    <card :title="$t('logs')" icon="file-text" no-body>
+      <template #header-buttons>
+        <b-button @click="shareLogs" variant="success">
+          <icon iname="cloud-upload" /> {{ $t('logs_share_with_yunopaste') }}
+        </b-button>
+      </template>
+
+      <b-button
+        v-if="moreLogsAvailable"
+        variant="white" class="w-100 rounded-0"
+        @click="$ref.view.fetchQueries()"
+      >
+        <icon iname="plus" /> {{ $t('logs_more') }}
+      </b-button>
+
+      <pre class="log"><code v-html="logs" /></pre>
+
+      <b-button @click="shareLogs" variant="success" class="w-100 rounded-0">
+        <icon iname="cloud-upload" /> {{ $t('logs_share_with_yunopaste') }}
+      </b-button>
+    </card>
+  </view-base>
+</template>
+
+<script>
+import api from '@/api'
+import { objectToParams } from '@/helpers/commons'
+import { readableDate } from '@/helpers/filters/date'
+
+export default {
+  name: 'ToolLog',
+
+  props: {
+    name: { type: String, required: true }
+  },
+
+  data () {
+    return {
+      // Log data
+      description: undefined,
+      info: {},
+      logs: undefined,
+      // Logs line display
+      numberOfLines: 25,
+      moreLogsAvailable: false
+    }
+  },
+
+  computed: {
+    queries () {
+      const queryString = objectToParams({
+        path: this.name,
+        filter_irrelevant: '',
+        with_suboperations: '',
+        number: this.numberOfLines
+      })
+      return ['logs/display?' + queryString]
+    }
+  },
+
+  methods: {
+    formatLogData (log) {
+      if (log.logs.length === this.numberOfLines) {
+        this.moreLogsAvailable = true
+        this.numberOfLines *= 10
+      } else {
+        this.moreLogsAvailable = false
+      }
+      this.description = log.description
+
+      const levels = ['ERROR', 'WARNING', 'SUCCESS', 'INFO']
+      this.logs = log.logs.map(line => {
+        for (const level of levels) {
+          if (line.includes(level + ' -')) {
+            return `<span class="alert-${level === 'ERROR'
+              ? 'danger'
+              : level.toLowerCase()}">${line}</span>`
+          }
+        }
+        return line
```

Comment:
```suggestion
      function escapeHtml (unsafe) {
        return unsafe
              .replace(/&/g, '&amp;')
              .replace(/</g, '&lt;')
              .replace(/>/g, '&gt;')
              .replace(/"/g, '&quot;')
              .replace(/'/g, '&#039;')
      }
      this.logs = log.logs.map(line => {
        for (const level of levels) {
          if (line.includes(level + ' -')) {
            return escapeHtml(`<span class="alert-${level === 'ERROR'
              ? 'danger'
              : level.toLowerCase()}">${line}</span>`)
          }
        }
        return escapeHtml(line)
```

Escape html in logs (is it the right way to do it?)

Reply:
```suggestion
      this.logs = log.logs.map(line => {
        const escaped = escapeHtml(line)
        for (const level of levels) {
          if (line.includes(level + ' -')) {
            return `<span class="alert-${level === 'ERROR'
              ? 'danger'
              : level.toLowerCase()}">${escaped}</span>`
          }
        }
        return escaped
```

We can escape the line indeed in case the logs contains HTML, but escaping the `<span class="alert-">` that allows highlighting will break it !

Also there is multiple potential problems (that existed in the Sammy version) :
- Some errors sent by the api contains HTML (sent as text)
- Some translations values contains HTML
Those are not escaped because it may contains formatting and links and comes from locales files, if the translations are carefully reviewed before merging it may be ok but [it is not recommended](https://kazupon.github.io/vue-i18n/guide/formatting.html#html-formatting). vue-i18n recommends to use [component interpolation](https://kazupon.github.io/vue-i18n/guide/interpolation.html) but this breaks sentences into multiple chunks so translators may not know what to do with it.